vaccine/states.py

```diff
@@ -254,3 +254,32 @@ async def display(self, message: Message):
                 ],
             },
         )
+
+
+class SectionedChoiceState(ChoiceState):
+    def __init__(self, *args, sections: dict, separator: Optional[str] = "", **kwargs):
```

Comment:
Having `sections` be a `dict` could be a bit troublesome, since a `dict` doesn't have any ordering, and we probably want a fixed order for the sections. Maybe a `Sequence[Tuple[str, Sequence[Choice]]]` is better, eg. `[("section1", [Choice(...), ...]), ("section2", [Choice(...), ...])]`